SSZ-QL: custom Generic Merkle Proofs building the tree and collecting the hashes in one sweep (#16177)
prysmaticlabs/prysm: changelog/fernantho_connect-merkle-proofs-with-endpoint.md

### Added

- Added new proofCollector type to ssz-query

> - Integrated Merkle proofs into ssz-query API endpoints

Reviewer: ```suggestion
```

> 
> - Added New ProofCollector type. It collects sibling hashes and leaves needed for Merkle proof during the merkleization.
>   - ProofCollector uses goroutines for vector merkleization. ProofCollector is thread-safe.
>   - ProofCollector returns `fastssz.Proof` enabling the usage of `fastssz` `Verify` function.
> - Integrated Merkle proofs into ssz-query API endpoints
> - Added testing covering the production of Merkle proof from Phase0 beacon state and benchmarked against real Hoodi beacon state (Fulu version)

Reviewer 2: ```suggestion
### Added

- Added new ProofCollector type to ssz-query
- Integrated Merkle proofs into ssz-query API endpoints

### Ignored
- Added testing covering the production of Merkle proof from Phase0 beacon state and benchmarked against real Hoodi beacon state (Fulu version)
```
I don\t think explaining how the `ProofCollector` works should be put in the changelog. I see you have this information in the godoc of the `ProofCollector` struct, which is the ideal place for this.
